core/src/main/java/com/graphhopper/routing/util/VehicleEncodedValues.java

```diff
@@ -108,11 +100,10 @@ public static VehicleEncodedValues car(PMap properties) {
         double speedFactor = properties.getDouble("speed_factor", 5);
         boolean speedTwoDirections = properties.getBool("speed_two_directions", false);
         int maxTurnCosts = properties.getInt("max_turn_costs", properties.getBool("turn_costs", false) ? 1 : 0);
-        double maxSpeed = properties.getDouble("max_speed", CAR_MAX_SPEED);
         BooleanEncodedValue accessEnc = new SimpleBooleanEncodedValue(getKey(name, "access"), true);
         DecimalEncodedValue speedEnc = new DecimalEncodedValueImpl(getKey(name, "average_speed"), speedBits, speedFactor, speedTwoDirections);
         DecimalEncodedValue turnCostEnc = maxTurnCosts > 0 ? TurnCost.create(name, maxTurnCosts) : null;
-        return new VehicleEncodedValues(name, accessEnc, speedEnc, null, null, turnCostEnc, speedEnc.getNextStorableValue(maxSpeed), true, false);
+        return new VehicleEncodedValues(name, accessEnc, speedEnc, null, null, turnCostEnc, true, false);
```

Comment:
The improvement here is that the 'maximum speed' no longer needs to be set for the FlagEncoder and the VehicleTagParser, but only the parser and then the maximum is taken from the speed EV (#2614)